Demo/Demo/Networking/DemoMerchantAPI.swift

```diff
@@ -67,4 +70,36 @@ final class DemoMerchantAPI {
     private func buildBaseURL(with endpoint: String) -> URL? {
         URL(string: DemoSettings.environment.baseURL + endpoint)
     }
+    
+    public func getAccessToken(clientId: String, environment: PaymentsCore.Environment) async -> String?{
+        guard let token = self.accessToken else{
+            self.accessToken = await fetchAccessToken(clientId: clientId, environment: environment)
+            return self.accessToken
+        }
+        return token
+    }
+    
+    private func fetchAccessToken(clientId: String, environment: PaymentsCore.Environment) async -> String? {
```

Comment:
Is there a reason we can't have this logic in the `getAccessToken` function? It seems like all `getAccessToken` does is call into this function and sets `self.accessToken` to the access token. I'm not fully understanding why we need a variable for this and cannot just condense this down to 1 function returning either the access token or throwing an error.

Reply:
here `getAccessToken` will check variable if it exists, if not makes a call to fetch token, idea is not to repeat the call, get it save once and use it